server/middleware/token.js

```diff
@@ -101,24 +111,26 @@ function token(options) {
     assert(typeof TokenModel === 'function',
       'loopback.token() middleware requires a AccessToken model');
 
+    var err = null;
     if (req.accessToken !== undefined) {
       if (!enableDoublecheck) {
         // req.accessToken is defined already (might also be "null" or "false") and enableDoublecheck
         // has not been set --> skip searching for credentials
-        rewriteUserLiteral(req, currentUserLiteral);
-        return next();
+        err = rewriteUserLiteral(req, currentUserLiteral);
+        return next(err);
```

Comment:
While this solution definitely works, it's quite an unusual pattern.

In Node.js world, the convention for functions is to accept a callback argument and then pass any errors to that callback.

```js
rewriteUserLiteral(req, currentUserLiteral, next);
```